lib/publiccloud/ec2.pm

```diff
@@ -130,17 +160,17 @@ sub upload_img {
           . "--machine '" . $img_arch . "' "
           . "-n '" . $self->prefix . '-' . $img_name . "' "
           . "--virt-type hvm --sriov-support "
-          . ($img_name =~ /byos/i ? '' : '--use-root-swap ')
+          . (is_byos() ? '' : '--use-root-swap ')
```

Comment:
@cfconrad  before we were relying on filename , now we switching to flavor name ( see `is_byos` implementation ) do you see any potential issue here ?

Reply:
AFAIK `is_byos()` is checking the `FLAVOR` variable. And we have always that upload Job dependency per flavor.
So we do not upload a `!BYOS` in a `FLAVOR=*BYOS*` job. With this, I don't see a potential problem!

If you use `openqa-clone-job` and overwrite only the `PUBLIC_CLOUD_IMAGE_LOCATION` variable and you get the above described missmatch, theny you have a problem. But this is only in "debugging" and the "debugger" should know what he is doing.

=> No :)

Reply:
But I'm missing the upload jobs for QAM? @grisu48 can you give some light. Just checked this https://openqa.suse.de/tests/5275015#settings and `PUBLIC_CLOUD_IMAGE_ID` and `PUBLIC_CLOUD_IMAGE_LOCATION` are empty. Also no job dependency.

Reply:
AFAIK they stop upload images and just install updates on latest image available using pint to get info which one . So QEM (QAM)  not affected by this flow anymore .

Reply:
thx for explanation